Replace wordmarks in chef-server-ctl (#1949, first pass)

jgitlin-p21 opened the pull request:
### Description

Replace "Wordmarks" (trademarked strings) in output with constants from `ChefUtils::Dist`, allowing wordmarks to be removed / replaced for community builds

### Issues Resolved

[Make Chef Server distributable under a configurable name - wordmarks in outputs #1949](https://github.com/chef/chef-server/issues/1949)

(Not resolved, this is a first pass)

### Check List

- [ ] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [ ] PR title is a worthy inclusion in the CHANGELOG

jgitlin-p21 commented:
Just rebased off `main` and pushed an updated branch.

jgitlin-p21 commented on src/chef-server-ctl/plugins/rotate_credentials.rb:
```diff
@@ -224,7 +225,7 @@ def restore_secrets_file(backup_file)
 
 def secrets_file_path
-  "/etc/opscode/private-chef-secrets.json"
+  "/etc/opscode/private-#{ChefUtils::Dist::Infra::SHORT}-secrets.json"
```
Good question. It was intended but I am unsure if this will break when the Cinc project replaces this wordmark...

ramereth replied:
This should also fix the `opscode` reference with `LEGACY_CONF_DIR`.

jgitlin-p21 replied:
> This should also fix the `opscode` reference with `LEGACY_CONF_DIR`.

Fixed @ramereth -- https://github.com/chef/chef-server/pull/2808/files#diff-5394b6ccc8748c324c60783e1004d761a3192133dc8043ebedf36b44de17f262R38

jgitlin-p21 commented on src/chef-server-ctl/plugins/gather-logs.rb:
```diff
@@ -2,11 +2,10 @@
     exit 1
   end
-  run_command("/opt/opscode/bin/gather-logs")
+  run_command("/opt/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/bin/gather-logs")
```
Yes, but maybe there's a better constant to use here? 

```LEGACY_CONF_DIR = "opscode"```

Looking to enable the Cinc project to replace `/etc/opscode` with `/etc/cinc-project`

ramereth replied:
Yes this is the correct one to use.

lbakerchef replied:
Maybe I don't have the proper context to what's going on, or "the big picture."  But the general theme here is, if you change a reference to a directory, things will break unless you also do other things to fix up that directory reference (rename the directory, make sure the system knows the directory has been renamed, etc.).

ramereth replied:
See my comment above please.

PrajaktaPurohit replied:
yeah we would have liked to get off the opscode directories - but for now the LEGACY_CONF_DIR seems the right path forward.

jgitlin-p21 commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -288,7 +289,7 @@ log_level = if ENV["CSC_LOG_LEVEL"]
 ChefServerCtl::Log.level = log_level
 
-ctl = Omnibus::ChefServerCtl.new(cmd_name, true, "Chef Server")
+ctl = Omnibus::ChefServerCtl.new(cmd_name, true, "#{ChefUtils::Dist::Server::PRODUCT}")
```
Yes. Note that this isn't an _exact_ replacement, the result becomes `Chef Infra Server` instead of `Chef Server`

If we prefer an exact replacement I could use:

```"#{ChefUtils::Dist::Infra::Short.capitalize} Server"```

Or we could add a new constant to `ChefUtils::Dist::Server`

ramereth replied:
IMHO it is likely preferred to change it to `Chef Infra Server` so this is the correct constant.

PrajaktaPurohit replied:
yeah it is only used as a display sing in https://github.com/chef/omnibus-ctl/blob/main/lib/omnibus-ctl.rb So it should be fine to use Chef Infra Server instead of Chef Server. Thank you for the fix!

jgitlin-p21 commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -28,13 +29,13 @@ module Omnibus
     end
 
-    CREDENTIAL_ENV = "CHEF_SECRETS_DATA".freeze
+    CREDENTIAL_ENV = "#{ChefUtils::Dist::Infra::SHORT.upcase}_SECRETS_DATA".freeze
```
@lbakerchef I did each one of these manually. I used IntelliJ, searched for all instances of `chef` inside strings, and then replaced each one by hand.

This appeared to me to be an environment variable used for configuration, so I assumed (perhaps wrongly) that it should use the product name to be compliant with the trademark policy. If this was a mistake and it's OK for Cinc Server to use `CHEF_SECRETS_DATA` here, then I will remove this change

lbakerchef replied:
@jgitlin-p21 No worries, whether `sed` (I use it all the time) or IntelliJ or whatever :)  My only point was, this looks like something that will break the code, so I thought it was unintentional.  Someone like @PrajaktaPurohit might know for sure looking at this.  Unless you've already tested this and know it works?

jgitlin-p21 replied:
It shouldn't break _your_ code because `ChefUtils::Dist::Infra::SHORT.upcase == "CHEF"` so this is not a change.

It might require further changes to _Cinc_ because this will become `"CINC_SECRETS_DATA"`, but in that case I just need to find any other cases where `"CHEF_SECRETS_DATA"` is used and make the same replacement...

PrajaktaPurohit replied:
Thank you @jgitlin-p21! @lbakerchef https://github.com/chef/chef/blob/main/chef-utils/lib/chef-utils/dist.rb#L43 will do the right thing and not make any changes to CREDENTIAL_ENV for Chef builds.

jgitlin-p21 commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -28,13 +29,13 @@ module Omnibus
         @credentials = Veil::CredentialCollection::ChefSecretsEnv.new(var_name: CREDENTIAL_ENV)
       else
-        secrets_file = ENV["SECRETS_FILE"] || "/etc/opscode/private-chef-secrets.json"
+        secrets_file = ENV["SECRETS_FILE"] || "/etc/opscode/private-#{ChefUtils::Dist::Infra::SHORT}-secrets.json"
```
Yes, but maybe this was me going too far. I am not sure if configuration files like this need to have the trademark/workmarks removed from their path names?

ramereth replied:
IMO this should stay and also fix the `opscode` reference to use the constant for `LEGACY_CONF_DIR`

lbakerchef replied:
These are directory names, so if you change those, things will break unless you also change the directories themselves.

PrajaktaPurohit replied:
From what I understand https://github.com/chef/chef/blob/main/chef-utils/lib/chef-utils/dist.rb#L43 will make it so that there are no changes for Chef builds.

jgitlin-p21 commented:
@lbakerchef here's the result of `chef-server-ctl test`, please let me know if there's other tests that I should do:

https://gist.github.com/jgitlin-p21/a38c2ff1c77ca4bdd13548438d9cc196

jashaik commented:
Pipeline Information-
Adhoc Build - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3215

ramereth commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -75,13 +76,13 @@ module Omnibus
           path = nil
         else
-          path = Time.now.strftime("/root/%FT%R-chef-server-manual-postgresql-cleanup.sql")
+          path = Time.now.strftime("/root/%FT%R-#{ChefUtils::Dist::Infra::SHORT}-server-manual-postgresql-cleanup.sql")
```
Yes

jgitlin-p21 commented:
> @jgitlin-p21 See @jashaik 's tests above. Those are the main tests we normally run (well one is a build, the other is a test). Looks like there's a few failures on the umbrella tests.

@lbakerchef I am not able to access those links because I am not a Chef employee. Can you tell me how to run the tsts locally, and/or what the failure was?

jashaik commented:
> > @jgitlin-p21 See @jashaik 's tests above. Those are the main tests we normally run (well one is a build, the other is a test). Looks like there's a few failures on the umbrella tests.
> 
> @lbakerchef I am not able to access those links because I am not a Chef employee. Can you tell me how to run the tsts locally, and/or what the failure was?

@jgitlin-p21 There are some issues while upgrade, I suggest you to test all the commands of chef-server-ctl manually in your chef-server setup and publish the results.
Please find the upgrade failure screenshot(chef-server 14.8.7 to your changes)
![Screenshot 2021-09-27 at 10 53 37 AM](https://user-images.githubusercontent.com/76031665/134849600-1468a543-9d08-4e9d-85aa-d534e30960a9.png)

jgitlin-p21 commented:
Thank you! I can address that, and will update the PR with a fix.

PrajaktaPurohit commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -6,6 +6,7 @@ require "veil"
 require "chef_server_ctl/log"
 require "chef_server_ctl/config"
 require "license_acceptance/acceptor"
+require "chef-utils/dist"
```
nice! we are on "v16.13.16" for server. Looks like this shipped in 16.5.55

PrajaktaPurohit commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -75,13 +76,13 @@ module Omnibus
           connection.exec(last)
         end
-        puts "Chef Server databases and roles have been successsfully deleted from #{postgres["vip"]}"
+        puts "#{ChefUtils::Dist::Server::PRODUCT} databases and roles have been successfully deleted from #{postgres['vip']}"
```
Nice! It should be Chef Infra Server not Chef Server. (ref: https://github.com/chef/chef/blob/main/chef-utils/lib/chef-utils/dist.rb#L103)

PrajaktaPurohit commented on src/chef-server-ctl/plugins/filtered-dump.rb:
```diff
@@ -1,6 +1,6 @@
 
 
-add_command_under_category "filtered-dump", "Debug Tools", "Generate a filtered dump of indexable Chef Objects for all organizations. Top-level data is captured; only object name is captured from object json" do
+add_command_under_category "filtered-dump", "Debug Tools", "Generate a filtered dump of indexable #{ChefUtils::Dist::Server::PRODUCT} Objects for all organizations. Top-level data is captured; only object name is captured from object json" do
```
Since these are not Chef Infra Server only objects per se - It might be ok to use the replacement for Chef instead of 'Chef Infra Server'

jgitlin-p21 replied:
That makes sense, thank you! Fixing this up now.

PrajaktaPurohit commented on src/chef-server-ctl/plugins/gather-logs.rb:
```diff
@@ -2,11 +2,10 @@
 #
-
-add_command_under_category "gather-logs", "general", "Create a tarball of recent logs and system information for Chef Support", 2 do
+add_command_under_category "gather-logs", "general", "Create a tarball of recent logs and system information for #{ChefUtils::Dist::Server::PRODUCT} Support", 2 do
```
Similar to one above, this should be appropriate replacement for Chef instead of replacing with Chef Infra Server, if possible.

jgitlin-p21 replied:
That makes sense. Now that I think about this... should we _not_ replace this word and actually intentionally leave it as `Chef Support`? Because, in this case it is actually talking about getting support from Progress/Chef, and if we were to replace this with E.G. "Cinc Support" (for the Cinc project) that doesn't make sense.

Up to you and @tas50, whatever you prefer here I'll do! 😄

PrajaktaPurohit commented on src/chef-server-ctl/plugins/upgrade.rb:
```diff
@@ -22,19 +23,19 @@ def parse(args)
       end
 
-      opts.on("-o", "--org-name [name]", String, "The name of the Chef 12 organization to be created. It must begin with a lower case letter or digit; can only have lower case letters, digits, hyphens, and underscores and must be between 1 and 255 characters long (Will ask interactively if not passed).") do |n|
+      opts.on("-o", "--org-name [name]", String, "The name of the #{ChefUtils::Dist::Infra::Org} 12 organization to be created. It must begin with a lower case letter or digit; can only have lower case letters, digits, hyphens, and underscores and must be between 1 and 255 characters long (Will ask interactively if not passed).") do |n|
```
Does not look like there is ChefUtils::Dist::Infra::Org, did you intend to use ChefUtils::Dist::Server::PRODUCT instead?

jgitlin-p21 replied:
Thanks! Yeah I don't know what I was trying to do here, but I just changed (and will shortly commit) this to: `ChefUtils::Dist::Infra::SHORT.capitalize`

jgitlin-p21 replied:
Unless you prefer `ChefUtils::Dist::Server::PRODUCT `

PrajaktaPurohit replied:
Any changes to chef-utils will not reflect in server unless the version of chef client is updated in server. So perhaps for now we could use what is available if that seems appropriate?

jgitlin-p21 replied:
Do you care if the wording changes from `The name of the Chef 12 organization` to `The name of the Chef Infra Server 12 organization`?

Using `ChefUtils::Dist::Infra::SHORT.capitalize` feels a bit awkward, but it does leave the text the same...

jgitlin-p21 commented on src/chef-server-ctl/plugins/backup.rb:
```diff
@@ -48,21 +48,21 @@
     end
 
-    opts.on("-c", "--cleanse", "Agree to cleansing all existing state during a restore.  THIS WILL COMPLETELY REMOVING EXISTING CHEF DATA") do
+    opts.on("-c", "--cleanse", "Agree to cleansing all existing state during a restore.  THIS WILL COMPLETELY REMOVING EXISTING #{ChefUtils::Dist::Server::PRODUCT.upcase} DATA") do
```
Fixed! https://github.com/chef/chef-server/pull/2808/files#diff-71592e0d65e422c11afe8842174b5dd6b35320f8f4e47ad7a3c606abd7b4cc2cR65

jgitlin-p21 commented on src/chef-server-ctl/plugins/maintenance.rb:
```diff
@@ -51,7 +51,7 @@ def validate_ip(ip)
 end
 
-add_command_under_category "maintenance", "general", "Control the maintenance mode on the Chef Infra Server", 2 do
+add_command_under_category "maintenance", "general", "Control the maintenance mode on the #{ChefUtils::Dist::Server::PRODUCT}", 2 do
```
Fixed! https://github.com/chef/chef-server/pull/2808/files#diff-8819384772f83b99bce9d96126124376907692527c856648c2c48cf31eda81fbR54

jgitlin-p21 commented:
Updated based on PR feedback, rebased off `main` and force-pushed. Re-running a build now and after that's completed I will re-run all tests as well as test every plugin (including the upgrade command which failed on my first attempt)

jgitlin-p21 commented:
[Here's a log](https://gist.github.com/jgitlin-p21/82371403cff82e9551acaf498078bd2a) showing that `chef-server-ctl upgrade` works now. I am running `chef-server-ctl test --all` now and will post the results of that when completed!

jgitlin-p21 commented:
And [here's a log](https://gist.github.com/jgitlin-p21/c66ff1b3b848138bb89329a330fa05ed) showung the upgrade and tests passing on the build newly rebased from `main`

jashaik commented:
New pipeline information -
Adhoc - https://buildkite.com/chef/chef-chef-server-main-omnibus-adhoc/builds/3246
Umbrella - https://buildkite.com/chef/chef-umbrella-main-chef-server/builds/724

tas50 commented on src/chef-server-ctl/bin/chef-server-ctl:
```diff
@@ -28,13 +29,13 @@ module Omnibus
         @credentials = Veil::CredentialCollection::ChefSecretsEnv.new(var_name: CREDENTIAL_ENV)
       else
-        secrets_file = ENV["SECRETS_FILE"] || "/etc/opscode/private-chef-secrets.json"
+        secrets_file = ENV["SECRETS_FILE"] || "/etc/#{ChefUtils::Dist::Org::LEGACY_CONF_DIR}/private-#{ChefUtils::Dist::Infra::SHORT}-secrets.json"
```
Is short lowercase?

tas50 commented on src/chef-server-ctl/plugins/upgrade.rb:
```diff
@@ -22,19 +23,19 @@ def parse(args)
       end
 
-      opts.on("-o", "--org-name [name]", String, "The name of the Chef 12 organization to be created. It must begin with a lower case letter or digit; can only have lower case letters, digits, hyphens, and underscores and must be between 1 and 255 characters long (Will ask interactively if not passed).") do |n|
+      opts.on("-o", "--org-name [name]", String, "The name of the #{ChefUtils::Dist::Infra::SHORT.capitalize} 12 organization to be created. It must begin with a lower case letter or digit; can only have lower case letters, digits, hyphens, and underscores and must be between 1 and 255 characters long (Will ask interactively if not passed).") do |n|
```
```suggestion
      opts.on("-o", "--org-name [name]", String, "The name of the #{ChefUtils::Dist::Infra::SHORT.capitalize} organization to be created. It must begin with a lower case letter or digit; can only have lower case letters, digits, hyphens, and underscores and must be between 1 and 255 characters long (Will ask interactively if not passed).") do |n|
```
